cms/djangoapps/contentstore/utils.py

```diff
@@ -1523,6 +1523,46 @@ def format_in_process_course_view(uca):
     return active_courses, archived_courses, in_process_course_actions
 
 
+def get_course_context_v2(request):
+    """Get context of the homepage course tab from the Studio Home."""
+
+    from cms.djangoapps.contentstore.views.course import (
```

Comment:
Why is this import inside this function? Can we add a comment to explain?

Reply:
I moved it outside the function along with the other imports and got:
```
cms-1  | Traceback (most recent call last):
cms-1  |   File "./manage.py", line 103, in <module>
cms-1  |     startup.run()
cms-1  |   File "/openedx/edx-platform/cms/startup.py", line 20, in run
cms-1  |     django.setup()
cms-1  |   File "/openedx/venv/lib/python3.8/site-packages/django/__init__.py", line 24, in setup
cms-1  |     apps.populate(settings.INSTALLED_APPS)
cms-1  |   File "/openedx/venv/lib/python3.8/site-packages/django/apps/registry.py", line 124, in populate
cms-1  |     app_config.ready()
cms-1  |   File "/openedx/edx-platform/cms/djangoapps/contentstore/apps.py", line 23, in ready
cms-1  |     from .signals import handlers  # pylint: disable=unused-import
cms-1  |   File "/openedx/edx-platform/cms/djangoapps/contentstore/signals/handlers.py", line 19, in <module>
cms-1  |     from cms.djangoapps.contentstore.courseware_index import (
cms-1  |   File "/openedx/edx-platform/cms/djangoapps/contentstore/courseware_index.py", line 15, in <module>
cms-1  |     from cms.djangoapps.contentstore.course_group_config import GroupConfiguration
cms-1  |   File "/openedx/edx-platform/cms/djangoapps/contentstore/course_group_config.py", line 12, in <module>
cms-1  |     from cms.djangoapps.contentstore.utils import reverse_usage_url
cms-1  |   File "/openedx/edx-platform/cms/djangoapps/contentstore/utils.py", line 88, in <module>
cms-1  |     from cms.djangoapps.contentstore.views.course import (
cms-1  |   File "/openedx/edx-platform/cms/djangoapps/contentstore/views/__init__.py", line 3, in <module>
cms-1  |     from .assets import *
cms-1  |   File "/openedx/edx-platform/cms/djangoapps/contentstore/views/assets.py", line 6, in <module>
cms-1  |     from cms.djangoapps.contentstore.asset_storage_handlers import (
cms-1  |   File "/openedx/edx-platform/cms/djangoapps/contentstore/asset_storage_handlers.py", line 35, in <module>
cms-1  |     from .utils import reverse_course_url, get_files_uploads_url, get_response_format, request_response_format_is_json
cms-1  | ImportError: cannot import name 'reverse_course_url' from partially initialized module 'cms.djangoapps.contentstore.utils' (most likely due to a circular import) (/openedx/edx-platform/cms/djangoapps/contentstore/utils.py)
```
I have already added the comment explaining why it needs to be done.